win32ss/user/user32/windows/cursoricon.c

```diff
@@ -2622,16 +2882,31 @@ HICON WINAPI CreateIconFromResourceEx(
             pbIconBits = (PBYTE)pt;
         }
 
-        if (!CURSORICON_GetCursorDataFromBMI(&cursorData, (BITMAPINFO*)pbIconBits))
+        isAnimated = FALSE;
+
+        /* Try to load BMP icon */
+        if (!CURSORICON_GetCursorDataFromBMI(&cursorData, (PBITMAPINFO)pbIconBits))
         {
-            ERR("Couldn't fill the CURSORDATA structure.\n");
-            if (ResHandle)
-                FreeResource(ResHandle);
-            return NULL;
+            /* Convert PNG raw data to BMP icon if the icon was PNG icon */
+            pbBmpIcon = CURSORICON_ConvertPngToBmpIcon(pbIconBits, cbIconBits, &BmpIconSize);
+            if (!pbBmpIcon)
+                goto end; /* Not PNG icon or failed */
```

Comment:
BUGFIX:
```suggestion
                return NULL; /* Not PNG icon or failed */
```
You may verify that there is no cleanup code that need to be run there.
(Also, this avoid taking the "end" code path, where hIcon would be left uninitialized, and this would trigger a runtime check error.)

Reply:
Changed. Thanks.